### PR TITLE
Add mutation support for Android Layout files

### PR DESCRIPTION
**Pull request by kkoser**
Changes to Android layout files are a common situation to profile, especially when using Android Data binding, which
will trigger its own compiler. This PR adds support for this form of mutations to the profiler. It supports mutating
both normal layout files and data binding layouts, which have a special format.

This is done by inserting a non-visible view into the bottom of the layout with a random ID. The ID is important
as it guanartees that the API of the data binding model for this layout will also change.

The one thing not supported currently is mutating a layout file with a single element inside it, as in that case there
is nowhere for a fake element to be inserted. Likely the solution to this is to make the fake element a wrapper layout,
but I don't see much use for that use case right now, so I left it out.

### CODE REVIEW

File: src/main/java/org/gradle/profiler/mutations/ApplyChangeToAndroidLayoutFileMutator.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.io.File;
+
+public class ApplyChangeToAndroidLayoutFileMutator extends AbstractFileChangeMutator {
+    public ApplyChangeToAndroidLayoutFileMutator(File sourceFile) {
+        super(sourceFile);
+    }
+
+    @Override
+    protected void applyChangeTo(StringBuilder text) {
+        // We need to handle both databinding layouts, which end with </layout>, and non, which can end with any layout item closing tag
```
Comment (oehme) [inline src/main/java/org/gradle/profiler/mutations/ApplyChangeToAndroidLayoutFileMutator.java]: Can you replace all these inline comments with a class-level comment describing what this does instead?

**Comment by kkoser** [comment]
Thanks @oehme for the quick response! I added some class level comments, and also added the flag in the README since I realized I forgot that part :) LMK if there's anything else I can do to improve it. Love the tool!

File: src/main/java/org/gradle/profiler/mutations/ApplyChangeToAndroidLayoutFileMutator.java (hunk 1 above)
Comment (oehme) [reply src/main/java/org/gradle/profiler/mutations/ApplyChangeToAndroidLayoutFileMutator.java]: I meant to also remove the inline comments, as they go stale and are redundant with the method names :)

Comment (kkoser) [reply src/main/java/org/gradle/profiler/mutations/ApplyChangeToAndroidLayoutFileMutator.java]: Ah sorry I misunderstood, let me update

**Comment by oehme** [comment]
Thank you very much! :)

**Pull request merged**